### DISCUSSION TITLE
It's really craizy. getStaticProps still can't use on _app.js

### DISCUSSION THREAD

**Discussion by makovez** [vercel/next.js#24764] [Help]
### What version of Next.js are you using?

no need

### What version of Node.js are you using?

no need

### What browser are you using?

no need

### What operating system are you using?

no need

### How are you deploying your application?

no need

### Describe the Bug

no need

### Expected Behavior

good

### To Reproduce

It's really frustrating, this makes me really think to trash away the whole project built with nextjs.

getStaticProps can't STIIILLL be used in _app.js . WHY?

It doesen't make any sense. Imagine you need to fetch many STATIC STUFF. which is ..**. STATIC** so it would make really sense to have those STATIC DATA ready at build time. Why is still not possible? why do I need to fetch this for every page when i can fetch from app and pass as props to any component?

But look at my scenario,
I have a navbar (which is a component) and I can't find any way to feed it at build time with static data as only page can be feed with getStaticProps. There is really no way. WHY? 

I need to feed static data at build time on my navbar, how can i do that?

ps. the Q/A page is FULL of this issues which don't seem to be answered. that's why i wrote here

**Comment by makovez** [comment]
I can't feed because navbar is not a page. It is a component unders
components folder.

Il lun 3 mag 2021, 20:38 Bruno Barros ***@***.***> ha scritto:

**Reply by bobarros** [reply]
Sorry that I deleted it. Thought you hadn't seen it.

I have one layout wrap for every page. It includes Navbar, Footer, Dark themes, and any other things shared across the whole website. 

I need to use getStaticProps on every page and it's less than ten models, though. But usually, it's already the case that every page has some getStaticProps for some other reason. The way you put it gave me the impression it was impossible.

**Comment by makovez** [comment]
So you have a component with all the shared components inside and you put
that on every page that uses it?

And then call getStaticProps on every page.... Ok.

But I think it would be still very nice to have it on _app.js

Il lun 3 mag 2021, 20:50 Bruno Barros ***@***.***> ha scritto:
